### ISSUE TITLE
When building from source, VERSION file is missing

### ISSUE THREAD

**Issue by jiweigert** [laurent-clouet/sheepit-client#299]
Hello,

I don't know if it's intentional,
but when I build sheepit-client from git-source (master), there is no VERSION file in the tree included.

That results on startup in a Console message:
``` Configuration::getJarVersion Failed to get version file```

and also in a wrong version number ```6.0.0``` instead of actual client version ```6.2038.0```, reported to the servers when connnected.

I hope this can be fixed?

**Comment by MCOfficer** [comment]
For the time being, you can use 
```bash
echo "6.$(svn info --xml 'https://github.com/laurent-clouet/sheepit-client/trunk' | xmllint --xpath 'string(/info/entry/@revision)' -).0" > ./resources/VERSION
```

derived from #267

**Comment by jiweigert** [comment]
Thank you, for the quick resonse.

so basically, 
```echo "6.$(svn info --xml 'https://github.com/laurent-clouet/sheepit-client/trunk' | xmllint --xpath 'string(/info/entry/@revision)' -).0" > ./resources/VERSION```
before 
```gradle4 ShadowJar```
right?

**Comment by MCOfficer** [comment]
> Thank you, for the quick resonse.
> 
> so basically,
> 
> ```
> echo "6.$(svn info --xml 'https://github.com/laurent-clouet/sheepit-client/trunk' | xmllint --xpath 'string(/info/entry/@revision)' -).0" > ./resources/VERSION 
> ```
> 
> before
> `gradle4 ShadowJar`
> right?

Exactly. Just make sure that the version is correct (`svn info` pulls from the latest master) to not fool the admins ;)

**Comment by jiweigert** [comment]
I would neeeeeeeever do that ;-) beside enabling opencl for my buildin Ryzen 5 3500U / Vega 8 GPU

**Comment by jiweigert** [comment]
Weird... with that VERSION file (svn version is 6.2039.0 inside VERSION) sheepit produce no output for the startup projects..

**Comment by Raimund58** [comment]
So, this can be closed?

**Comment by jiweigert** [comment]
Looks good so far, I had no issues since then, beside the usual,
non-related to this issue, once.

Yes, can be closed, Thank you.

Raimund58 <notifications@github.com> schrieb am Mi., 23. Sep. 2020, 09:25:
